### DISCUSSION TITLE
How does GPU usage detection work?

### DISCUSSION THREAD

**Discussion by Aashishkebab** [AutoDarkMode/Windows-Auto-Night-Mode#821] [Q&A]
I have two GPUs in my computer. Which does it use?

**Comment by Spiritreader** [comment]
It uses the "engtype_3D", "graphics" and "copy" windows performance counter for all of the GPU devices in your machine that support this metric. If you have two GPUs, it will use the counter that shows the heaviest load.

**Reply by AlmightyWings232** [reply]
Related to the question: What does GPU usage detection speed do? For example, what's the difference between fast and slow (are there advantages/disadvantages to picking either)

**Reply by Spiritreader** [reply]
> Related to the question: What does GPU usage detection speed do? For example, what's the difference between fast and slow (are there advantages/disadvantages to picking either)

The detection speed is essentially how many samples need to be below the configured threshold until switching is re-enabled.

The module has a tickrate of 60 seconds, meaning a new sample is drawn at this interval
